exchanges/okx/okx_websocket.go

```diff
@@ -0,0 +1,1848 @@
+	if err != nil {
+		return err
+	}
+	var a asset.Item
+	a, err = ok.GetAssetTypeFromInstrumentType(intermediate.Argument.InstrumentType)
+	if err != nil {
+		a = ok.GuessAssetTypeFromInstrumentID(intermediate.Argument.InstrumentID)
+	}
+	candleInterval := strings.TrimPrefix(intermediate.Argument.Channel, candle)
+	candlesticks := make([]stream.KlineData, len(response.Data))
+	for i := range response.Data {
+		candles, okay := (intermediate.Data[i]).([7]string)
+		if !okay {
+			return errIncompleteCandlestickData
+		}
+		timestamp, err := strconv.Atoi(candles[0])
+		if err != nil {
+			return err
+		}
+		candle := stream.KlineData{
+			Pair:      pair,
+			Exchange:  ok.Name,
+			Timestamp: time.UnixMilli(int64(timestamp)),
+			Interval:  candleInterval,
+			AssetType: a,
+		}
+		candle.OpenPrice, err = strconv.ParseFloat(candles[1], 64)
+		if err != nil {
+			return err
+		}
+		candle.HighPrice, err = strconv.ParseFloat(candles[2], 64)
+		if err != nil {
+			return err
+		}
+		candle.LowPrice, err = strconv.ParseFloat(candles[3], 64)
+		if err != nil {
+			return err
+		}
+		candle.ClosePrice, err = strconv.ParseFloat(candles[4], 64)
+		if err != nil {
+			return err
+		}
+		candle.Volume, err = strconv.ParseFloat(candles[5], 64)
+		if err != nil {
+			return err
+		}
+		candlesticks[i] = candle
+	}
+	ok.Websocket.DataHandler <- candlesticks
+	return nil
+}
+
+// wsProcessTickers handles the trade ticker information.
+func (ok *Okx) wsProcessTickers(data []byte) error {
+	var response WSTickerResponse
+	if err := json.Unmarshal(data, &response); err != nil {
+		return err
+	}
+	tickerPrices := make([]ticker.Price, len(response.Data))
+	for i := range response.Data {
+		a := response.Data[i].InstrumentType
+		if a == asset.Empty {
+			a = ok.GuessAssetTypeFromInstrumentID(response.Data[i].InstrumentID)
+		}
+		if !(ok.SupportsAsset(a)) {
+			return errInvalidInstrumentType
+		}
+		var c currency.Pair
+		var err error
+		c, err = ok.GetPairFromInstrumentID(response.Data[i].InstrumentID)
+		if err != nil {
+			return err
+		}
+		var baseVolume float64
+		var quoteVolume float64
+		if a == asset.Spot || a == asset.Margin {
+			baseVolume = response.Data[i].Vol24H
+			quoteVolume = response.Data[i].VolCcy24H
+		} else {
+			baseVolume = response.Data[i].VolCcy24H
+			quoteVolume = response.Data[i].Vol24H
+		}
+		tickerPrices[i] = ticker.Price{
+			ExchangeName: ok.Name,
+			Open:         response.Data[i].Open24H,
+			Volume:       baseVolume,
+			QuoteVolume:  quoteVolume,
+			High:         response.Data[i].High24H,
+			Low:          response.Data[i].Low24H,
+			Bid:          response.Data[i].BidPrice,
+			Ask:          response.Data[i].BestAskPrice,
+			BidSize:      response.Data[i].BidSize,
+			AskSize:      response.Data[i].BestAskSize,
+			Last:         response.Data[i].LastTradePrice,
+			AssetType:    a,
+			Pair:         c,
+			LastUpdated:  response.Data[i].TickerDataGenerationTime,
+		}
+	}
+	ok.Websocket.DataHandler <- tickerPrices
```

Comment:
Since this is changed to a slice, this needs to be handled in [websocket routine manager](https://github.com/thrasher-corp/gocryptotrader/blob/master/engine/websocketroutine_manager.go#L182)

`And all types that you have now changed to slices need to be handled and added to this switch case. `